fe/fe-core/src/main/java/com/starrocks/catalog/mv/MVTimelinessArbiter.java

```diff
@@ -224,4 +212,64 @@ protected void addEmptyPartitionsToRefresh(MvUpdateInfo mvUpdateInfo) {
             }
         });
     }
+
+    public Map<Table, Map<String, PCell>> syncBaseTablePartitions(MaterializedView mv) {
+        PartitionInfo partitionInfo = mv.getPartitionInfo();
+        if (partitionInfo.isUnPartitioned()) {
+            return null;
+        }
+        Map<Table, Map<String, PCell>> basePartitionNameToRangeMap = differ.syncBaseTablePartitionInfos();
+        if (CollectionUtils.sizeIsEmpty(basePartitionNameToRangeMap)) {
+            return null;
+        }
+        return basePartitionNameToRangeMap.keySet().stream()
+                .map(baseTable -> Maps.immutableEntry(baseTable, basePartitionNameToRangeMap.get(baseTable)))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    public PartitionDiff getChangedPartitionDiff(MaterializedView mv,
+                                                 Map<Table, Map<String, PCell>> basePartitionNameToRangeMap)  {
+        PartitionInfo partitionInfo = mv.getPartitionInfo();
+        try {
+            if (partitionInfo.isUnPartitioned()) {
+                return null;
+            }
+            PartitionDiffResult result = differ.computePartitionDiff(null,
+                    basePartitionNameToRangeMap);
+            if (result == null) {
+                logMVPrepare(mv, "Partitioned mv compute list diff failed");
+                return null;
+            }
+            return result.diff;
+        } catch (Exception e) {
+            LOG.warn("Materialized view compute partition difference with base table failed.", e);
```

Comment:
why not throw this exception

Reply:
If it can not be rewritten by mv, it can redirect to original query directly.